axiom/optimizer/ToGraph.cpp

```diff
@@ -1612,6 +1649,8 @@ PlanObjectP ToGraph::makeQueryGraph(
       return currentDt_;
     }
     case lp::NodeKind::kUnnest:
+      translateUnnest(*node.asUnchecked<lp::UnnestNode>());
```

Comment:
Shouldn't be something like this?

```suggestion
    case lp::NodeKind::kUnnest:
      if (!contains(allowedInDt, PlanType::kJoinNode)) {
        return wrapInDt(node);
      }
      translateUnnest(*node.asUnchecked<lp::UnnestNode>());
```